### ISSUE TITLE
Avoid rerendering tabs unnecessarily.

### ISSUE THREAD

**Issue by gerhean** [reposense/RepoSense#1382]
**What feature(s) would you like to see in RepoSense?**

Currently, tab info is passed from `store.js` to `main.js` to `v-authorship.js` and `v-zoom.js`.
**Suggestion:** to pass info directly from `store.js` to `v-authorship.js` and `v-zoom.js`. 

Each time tab information is updated, we destroy old component by the key, and the new component is generated based on the info in store instead of a prop passed in.
**Suggestion:** Remove the key and watch for info changes directly from the tabs.

**Is the feature request related to a problem?**

 Might be related to #967, #1167 and #967

From PR #1372:

![image](https://user-images.githubusercontent.com/16630400/102685587-d8474900-421c-11eb-8086-df6d1804abbf.png)

> 
> I agree that in the ideal design, the main.js can use the info in the store to generate the key instead of keeping a local copy. But we are using the current workflow as it is inherited from the old design. In our old design, the sidebar will be completely destroyed and re-rendered when updating, so we designed that we use a prop to control the sidebar because at that time, we haven't implemented the Vue store.
> 
> Now, you may try to see if it is possible to use the info in the store to generate the key instead of keeping a local copy and get the info from Vue store directly for sidebar components. So each time we update, we destroy old component by the key, and the new component is generated based on the info in store instead of a prop passed in.
> 
> And you may open a new PR for it to see if it works or not. This kind of PR needs to be very careful as it is touching the root design of our project.

**If possible, describe the solution**

Watch for specific changes in `info` using computed string as a key:

```
  watch: {
    authorshipKey() {
      this.initiate();
    },
```

```
 computed: {
    authorshipKey() {
      return `${this.info.author}|${this.info.repo}|${this.info.isMergeGroup}`;
    },
```

Calling `this.initiate();` is less destructive than rerendering the entire tab.

Not entirely sure which properties are important to watch however.

Proposed solution:
https://github.com/gerhean/RepoSense/tree/tabs-get-props-from-store

Will PR once PR #1372 is merged.
